cmd/srcd/cmd/defered.go

```diff
@@ -54,43 +72,60 @@ func (d *Defered) print(ch <-chan bool, done chan<- bool) {
 		return
 	}
 
-	logrus.Info(d.Msg)
+	d.logger.Infof(d.Msg)
 	if d.InputFn == nil {
 		done <- true
 		return
 	}
 
 	go func() {
 		for line := range d.InputFn(ch) {
-			logrus.Info(line)
+			d.logger.Infof(line)
 		}
 
 		done <- true
 	}()
 }
 
 func (d *Defered) spinner(msg string, stop <-chan bool, done chan<- bool) {
+	// skip if log level is not info or debug. We print directly into os.Stderr,
+	// so we cannot rely on log.Infof skipping the message
+	if log.DefaultFactory.Level != log.InfoLevel &&
+		log.DefaultFactory.Level != log.DebugLevel {
+		return
+	}
+
 	interval := d.SpinnerInterval
 	if interval == 0 {
 		interval = 200 * time.Millisecond
 	}
 
-	writer := logrus.StandardLogger().Out
+	// If the logger format is not text, or the output is not a terminal,
+	// do not print the spinner
+	if log.DefaultFactory.Format != log.TextFormat || !d.isTerminal() {
+		d.logger.Infof(d.Msg)
+
+		select {
+		case <-stop:
+			d.logger.Infof("%s, done", d.Msg)
+			done <- true
+			return
+		}
+	}
+
 	charset := []int{'⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏'}
 	i := 0
 	for {
 		select {
 		case <-stop:
-			logrus.Infof("%s, done", d.Msg)
+			d.logger.Infof("%s, done", d.Msg)
 			done <- true
 			return
 		default:
 			spinner := string(charset[i%len(charset)])
-			logrus.Infof("%s %s", d.Msg, spinner)
+			d.logger.Infof("%s %s", d.Msg, spinner)
 			time.Sleep(interval)
-			if writer == os.Stdout || writer == os.Stderr {
-				fmt.Fprint(writer, "\033[A")
-			}
+			fmt.Fprint(d.logWriter, "\033[A")
```

Comment:
Would it be possible to make `go-log` support these kinds of sequences? We could remove both [this](https://github.com/src-d/engine/pull/425/files#diff-0683c59e10bb2d32b77d1f043eb9b4dbR91) and `d.logWriter`.

Reply:
I don't think this should be part of go-log. In a way I feel we are misusing logs to show program output.

Reply:
fair enough, yup this isn't really a "real" log.